### PR TITLE
Add the input check for softmax_with_cross_entropy

### PR DESCRIPTION
**Pull request by yghstill**
### PR types

Function optimization
### PR changes

APIs
### Describe

Add the input check for `paddle.nn.functional.softmax_with_cross_entropy`.

### CODE REVIEW

File: python/paddle/nn/functional/loss.py [hunk 1]
```diff
@@ -260,6 +260,21 @@ def fluid_softmax_with_cross_entropy(
             # Tensor(shape=[1], dtype=float32, place=Place(gpu:0), stop_gradient=True,
             #        [1.15328646])
     """
+    input_dims = len(list(logits.shape))
+    if input_dims == 0:
```
Comment (phlrain) [inline python/paddle/nn/functional/loss.py]: 这个检查，能不能放到c++ 检查， 这个python检查对于调度性能影响比较大

Comment (yghstill) [reply python/paddle/nn/functional/loss.py]: done

**Review by phlrain — APPROVED** [review]
